cmd/bugzilla-mass-cloner/main.go

```diff
@@ -0,0 +1,153 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/test-infra/prow/bugzilla"
+	prowflagutil "k8s.io/test-infra/prow/flagutil"
+)
+
+type options struct {
+	prowflagutil.BugzillaOptions
+
+	fromRelease string
+	toRelease   string
+
+	dryRun bool
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+	fs.BoolVar(&o.dryRun, "dry-run", true, "Dry run for testing. Uses API tokens but does not mutate.")
+	fs.StringVar(&o.fromRelease, "from-release", "", "From which targeted release the bug will be cloned to")
+	fs.StringVar(&o.toRelease, "to-release", "", "To which release value the cloned bugs will hold")
+
+	o.BugzillaOptions.AddFlags(fs)
+
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		logrus.WithError(err).Fatal("could not parse input")
+	}
+
+	return o
+}
+
+func (o *options) validate() error {
+	if err := o.BugzillaOptions.Validate(o.dryRun); err != nil {
+		return err
+	}
+
+	if o.fromRelease == "" {
+		return fmt.Errorf("--from-release must be specified")
+	}
+
+	if o.toRelease == "" {
+		return fmt.Errorf("--to-release must be specified")
+	}
+
+	if o.fromRelease == o.toRelease {
+		return fmt.Errorf("--from-release and --to-release can't hold the same value")
+	}
+
+	return nil
+}
+
+type bugzillaClient interface {
+	SearchBugs(filters map[string]string) ([]*bugzilla.Bug, error)
+	CloneBug(bug *bugzilla.Bug, mutations ...func(bug *bugzilla.BugCreate)) (int, error)
+	GetClones(bug *bugzilla.Bug) ([]*bugzilla.Bug, error)
+}
+
+func (o options) getBugsByStatus(client bugzillaClient, statuses []string) ([]*bugzilla.Bug, error) {
+	var bugsToClone []*bugzilla.Bug
+
+	for _, status := range statuses {
+		bugs, err := client.SearchBugs(map[string]string{"target_release": o.fromRelease, "status": status})
+		if err != nil {
+			return nil, fmt.Errorf("couldn't search for bugs: %w", err)
+		}
+		bugsToClone = append(bugsToClone, bugs...)
+	}
+	return bugsToClone, nil
+}
+
+func hasCloneForTargetRelease(client bugzillaClient, bug *bugzilla.Bug, toRelease string) (bool, error) {
+	clonedBugs, err := client.GetClones(bug)
+	if err != nil {
+		return false, fmt.Errorf("couldn't get clones for bug with id %d: %w", bug.ID, err)
+	}
+
+	for _, clonedBug := range clonedBugs {
+		targetRelease := sets.NewString(clonedBug.TargetRelease...)
+		if targetRelease.Has(toRelease) {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func (o options) massCloneBugs(client bugzillaClient, bugs []*bugzilla.Bug) error {
+	var errs []error
+	for _, bug := range bugs {
+		if targetReleases := sets.NewString(bug.TargetRelease...); !targetReleases.Has(o.fromRelease) {
+			continue
+		}
```

Comment:
This check looks redundant? We use `fromRelease` (L73-L76) when searching for the bugs to clone so this method does not need to check anymore.

Reply:
You are right. This check satisfies only the tests. I'll fix this.